app/controllers/concerns/privacy_pass_supportable.rb

```diff
@@ -0,0 +1,32 @@
+module PrivacyPassSupportable
+  extend ActiveSupport::Concern
+
+  private
+
+  def setup_privacy_pass_challenge
+    return unless privacy_pass_enabled?
+
+    tokenizer = PrivacyPassTokenizer.new
+    tokenizer.register_challenge_for_redemption(session.id)
+    challenge = tokenizer.challenge_token
+    response.set_header("WWW-Authenticate", "PrivateToken challenge=#{challenge}, token-key=#{PrivacyPassTokenizer.issuer_public_key}")
+  end
+
+  def valid_privacy_pass_redemption?
+    return false unless privacy_pass_enabled?
+    return session[:redeemed_privacy_pass] unless session[:redeemed_privacy_pass].nil?
+
+    success = PrivacyPassRedeemer.call(request.headers["Authorization"], session.id)
+    session[:redeemed_privacy_pass] = success
+    success
+  end
+
+  def delete_privacy_pass_token_redemption
+    session.delete(:redeemed_privacy_pass)
+  end
+
+  def privacy_pass_enabled?
+    ld_context = LaunchDarkly::LDContext.with_key(self.class.name)
+    Rails.configuration.launch_darkly_client.variation("privacy_pass.enabled", ld_context, false)
```

Comment:
nit: let's call the flag `"gemcutter.privacy_pass.enabled"`. should we use the current controller & action as the key, so we can test by only turning it on for a single action, rather than every action?

Reply:
I updated the name.

I think using the current controller would be good. My only concern about using the action is that for every place we want it (sign up or sign in) we actually need it enabled in 2 actions. The `new` action presents the challenge and the `create` action handles redeeming the token.

Reply:
fair enough! I just think we'll need a `kind` on the LDContext of say `controller` so it doesn't look like a user